shapely/geometry/point.py

```diff
@@ -128,6 +136,10 @@ def bounds(self):
     # Coordinate access
 
     def _set_coords(self, *args):
+        warnings.warn(
+            "Setting the 'coords' to mutate a Geometry in place is deprecated,"
+            " and will not be possible any more in Shapely 2.0",
+            FutureWarning, stacklevel=2)
```

Comment:
@jorisvandenbossche what would you think about adding a new ShapelyDeprecationWarning as recommended in
https://www.python.org/dev/peps/pep-0565/#additional-use-case-for-futurewarning ?

> For library and framework authors that want to ensure their API compatibility warnings are more reliably seen by their users, the recommendation is to use a custom warning class that derives from DeprecationWarning in Python 3.7+, and from FutureWarning in earlier versions.